components/RuiExpandCollapse/src/RuiExpandCollapse.ts

```diff
@@ -0,0 +1,194 @@
+    if (this._collapseableEl) {
+      var sectionHeight = this._collapseableEl.scrollHeight;
+      this._collapseableEl.style.height = sectionHeight + 'px';
+
+      const transitionEndHandler = (): void => {
+        if (this._collapseableEl) {
+          this._collapseableEl.style.height = '';
+          this._collapseableEl.removeEventListener('transitionend', transitionEndHandler);
+        }
+      }
+  
+      this._collapseableEl.addEventListener('transitionend', transitionEndHandler);
+    }
+  }
+
+  /**
+   * Initialises the expand collapse logic and styling,
+   * once complete makes the expand collapse visible
+   */
+  private _initialiseExpandCollapse(): void {
+    if (this.shadowRoot) {
+      this._collapseableEl = this.shadowRoot.querySelector('.details-container');
+
+      if (this._collapseableEl) {
+        
+        // need to set height initially if closed without triggering animation
+        if (!this.open) {
+          this._collapseableEl.style.height = '0px';
+        }
+        
+        let expandCollapse: HTMLElement | null = this.shadowRoot.querySelector('.expand-collapse');
+        if (expandCollapse) {
+          expandCollapse.style.opacity = '1';
+        }
+      }
+    }
+  }
+  
+  /**
+   * After initial render initialise expand collapse logic. Because
+   * we are animating slotted content, we have to wait for the 
+   * slot to be mounted
+   */
+  public firstUpdated(): void {
+    if (this.shadowRoot) {
+      this._detailsSlotEl = this.shadowRoot.querySelector('#details-slot');
+
+      if (this._detailsSlotEl) {
+        // when the slotted content changes we initialise expand collapse
+        // we need to wait for this because the animation of heigh calc 
+        // will only work once the slot and it's content have mounted and rendered
+        this._detailsSlotEl.addEventListener('slotchange', (): void => {
+          this._initialiseExpandCollapse();
+         });
+      }
+    }
+  }
+
+  public updated(changedProperties: Map<string, RuiExpandCollapsePropertyType>): void {
+    changedProperties.forEach((oldValue: RuiExpandCollapsePropertyType, propName: string): void => {
+      // detect change in open prop and trigger animation as necessary
+      if (propName === 'open' && this._collapseableEl) {
+        // transition from closed to open
+        if (this.open && !oldValue) {
+          this._triggerExpandAnimation();
+        }
+         
+        // transition from open to closed
+        if (!this.open && oldValue) {
+          this._triggerCollapseAnimation();
+        }
+      }
+    });
+  }
+
+  /**
+   * Render method
+   */  
+  public render(): TemplateResult {
+    return html`
+      <section class=${`expand-collapse${this.open ?  ' is-open' : '' }`}>
+        <div @click="${this._handleClick}" class="summary-container" aria-expanded=${`${this.open ? 'true': 'false'}`}>
+          <slot name="summary-content"></slot>
+          <div class="icon-container"></div>
```

Comment:
Sorry if this has already been discussed, but I think this component becomes more reusable if you just remove this icon div and assume any icons will come as part of the summary-content slot.

This way it supports all sorts of designs (icon in front or no icon etc.)

Reply:
This is a really good point. 

I do like the idea of having icon styles upfront though.. Perhaps we need to think about how icons will work in this eco system

Reply:
Yeah definitely a good point - although if we follow a more generic approach users may find it frustrating when trying to target styles specifically to open or closed state of the component. Ideally the user would be able to show a different icon when it is open/closed without needing to externally control the component.

Reply:
Perhaps I'm missing something, but wouldn't the user just do what you're already doing? (i.e. target there own icon span with the `.is-open` class)

Reply:
@jeffdowdle The .is-open class is contained within the shadow dom so will not be targetable by the the slotted content as it is rendered outside the SD

Reply:
@jeffdowdle ...however I think I just found a way to expose the internal state, so my concern should no longer be a problem. @jennasalau If you're happy to remove the icon by default I can implement it now

Reply:
Oh I see, I think this has interesting implications for what kind of designs will be possible with this (we might be happy to take on those limitations, not sure). But I think it's outside the scope of this PR, so happy to leave this for now.

Reply:
@jeffdowdle i've recently made a small change so that the `open` attribute is 2-way-bound to the open internal state. So targeting whether it's open or closed would now be as simple as:
```css
rui-expand-collapse[open] .my-slotted-icon {
    /* whatever styles we want */
}
```